tests/test_autodiscover.py

```diff
@@ -1,186 +1,168 @@
-from pathlib import Path
-from unittest import mock
-from unittest.mock import MagicMock, patch
+import os
+import sys
+from unittest import TestCase, mock
 
-from django.template.engine import Engine
-from django.urls import include, path
+from django.conf import settings
 
-# isort: off
-from .django_test_setup import settings
-from .testutils import BaseTestCase
+from django_components import component, component_registry
+from django_components.autodiscover import _filepath_to_python_module, autodiscover, import_libraries
 
-# isort: on
+from .django_test_setup import setup_test_config
 
-from django_components import _filepath_to_python_module, autodiscover, component, component_registry
-from django_components.template_loader import Loader
 
-urlpatterns = [
-    path("", include("tests.components.urls")),
-]
+class _TestCase(TestCase):
+    @classmethod
+    def setUpClass(self) -> None:
+        component.registry.clear()
+        return super().setUpClass()
 
+    @classmethod
+    def tearDownClass(cls) -> None:
+        super().tearDownClass()
+        component.registry.clear()
```

Comment:
Why is this on the class level, and not on the test level? The component registry shouldn't be persisted between tests right? Also, I think just clearing on teardown should be enough if things are as they should...

Reply:
Good point, I used the class methods as I've seen them used before, but didn't question them. I've updated this throughout the tests